sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidCalciteSchemaModule.java

```diff
@@ -64,7 +64,7 @@ public void configure(Binder binder)
   }
 
   @Provides
-  @Singleton
+  @LazySingleton
```

Comment:
whats the reason for this change? Worthy of comment here you think, or nah?

Reply:
`@Singleton` should NEVER be used.  If it exists in any of our code we should remove it immediately.  That's the reason.  If anything it deserves a checkstyle validation.

Reply:
@imply-cheddar, thanks for your strident note. As it turns out, we can't _quite_ achieve what you ask. There is one other use of `@Singleton` in the code base, which I added in a previous PR. It handles the case in which Guice creates an instance explicitly. You indicated that you wanted only explicit bindings, but had to turn off that protection for some reason or other. The result is that, when Guice is used in tests, it will occasionally find itself implicitly instantiating some instances.

There are times when we need those implicit instances to be singletons. These implicit singletons are where the `@Singleton` or `@LazySingleton` tags are needed. (Again, the binding is "implicit" if there is no explicit binding, which means we didn't use `@LazySingleton` at that time.)

This is an example of a class that sometimes gets instantiated implicitly. So, we have to mark the class as `@[Lazy]Singleton`.

There is one other place in the code where changing `@Singleton` to `@LazySingleton` cases Guice to crash. Why? I didn't take the time to figure it out. That line _didn't_ change in this PR, but is the one we should be discussing.

To achieve @imply-cheddar 's dictate, we have to turn off implicit instantiation. By so doing, we eliminate the need to annotate classes.

Reply:
Please share where that LazySingleton thing is breaking?  It should be true that `LazySingleton` is the only singleton annotation that we use anywhere, if that's breaking for a reason, I'm very curious how it's breaking stuff and would like to help fix it.

Reply:
@imply-cheddar and I had an offline conversation to track down the `@LazySingleton` issues and he kindly provided a PR to resolve the issue.